homework9/tests/hw2_test.py

```diff
@@ -0,0 +1,17 @@
+import pytest
+
+from homework9.hw2 import SupressorClass, supressor
+
+
+@pytest.mark.parametrize(
+    ["sup", "err", "expression"],
+    [
+        (supressor, IndexError, "[][2]"),
+        (supressor, KeyError, 'dict()["key"]'),
+        (SupressorClass, IndexError, "[][2]"),
+        (SupressorClass, KeyError, 'dict()["key"]'),
+    ],
+)
+def test_supressor_success(sup, err, expression):
```

Comment:
You also should check if your suppressor did not catch other exceptions.

Reply:
I add these errors
```python
ALL_ERRORS = [
    AssertionError,
    AttributeError,
    EOFError,
    FloatingPointError,
    GeneratorExit,
    ImportError,
    IndexError,
    KeyError,
    KeyboardInterrupt,
    MemoryError,
    NameError,
    NotImplementedError,
    OSError,
    OverflowError,
    ReferenceError,
    RuntimeError,
    StopIteration,
    SyntaxError,
    IndentationError,
    TabError,
    SystemError,
    SystemExit,
    TypeError,
    UnboundLocalError,
    UnicodeError,
    ValueError,
    ZeroDivisionError
]
```